src/ReactCarousel.jsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react'
+import option from './Settings'
+import Arrow from './ReactArrow'
+import Dots from './ReactDots'
+import './style.css'
+
+export default class App extends React.Component {
+  constructor (props) {
+    super(props)
+
+    const children = this.props.children
+    const nbrSlides = children.length
+    const initialSlide = this.props.initialSlide || option.initialSlide
+    const carouselWidth = 100 * nbrSlides
+    const slideToScroll = this.props.slideToScroll || option.slideToScroll
+    const slideToShow = this.props.slideToShow || option.slideToShow
+    const slideWidth = nbrSlides * slideToShow
+    const gutter = slideToShow === 1 ? 0 : (this.props.gutter || option.gutter) / 100 * slideWidth * 100 / carouselWidth
+    const transitionDuration = this.props.transitionDuration || option.transitionDuration
+    const hideDots = this.props.hideDots || option.hideDots
+
+    this.state = { slideWidth, carouselWidth, gutter, slideToScroll, currentSlide: initialSlide, slideToShow, children, nbrSlides, transitionDuration, hideDots }
```

Comment:
comme pour le store de redux, je serais d'avis de mettre des valeurs par défaut. 
Rien que pour une review, tu ne sais pas ce que tu attends avec "slideWidth". Y a l'unité avec ? si oui, px ou em ?

Reply:
c'est du %

Reply:
au lieu de 
this.state = { slideWidth, carouselWidth, gutter, ... transitionDuration, hideDots }

Je te propose quelque de : 
- merge avec des const précédentes
- ordonner les states
- mettre des valeurs par défaut 

ex : 
```
this.state = {
      styled: {
        slideWidth: '1200px',
        carouselWidth: '100px',
        gutter :  '1em ',
        ...
      },
      currentSlide : 0,
      ...
      initialSlide : 0,
    }
```

Reply:
Je suis en train de faire ça plutot : 
```js
const {
      children,
      slideToScroll = option.slideToScroll,
      initialSlide = option.initialSlide,
      slideToShow = option.slideToShow,
      transitionDuration = option.transitionDuration,
      hideDots = option.hideDots
    } = this.props

this.state = {
      slideToScroll,
      currentSlide: initialSlide,
      slideToShow,
      children,
      nbrSlides,
      transitionDuration,
      hideDots
    }
```